caravel/assets/javascripts/explorev2/reducers/exploreReducer.js

```diff
@@ -2,30 +2,28 @@ import { defaultFormData, defaultOpts } from '../stores/store';
 import * as actions from '../actions/exploreActions';
 import { addToArr, removeFromArr, alterInArr } from '../../../utils/reducerUtils';
 
-const setFormInViz = function (state, action) {
-  const newFormData = Object.assign({}, state);
-  newFormData[action.key] = action.value;
+const setFormInViz = function (oldFd, action) {
+  const newFormData = Object.assign({}, oldFd);
+  // flip the value for checkbox where action.value is undefined
+  newFormData[action.key] = action.value ? action.value : (!oldFd[action.key]);
   return newFormData;
 };
 
-const setVizInState = function (state, action) {
+const setVizInState = function (oldViz, action) {
   switch (action.type) {
     case actions.SET_FORM_DATA:
```

Comment:
it looks like this only gets called `SET_FORM_DATA` action handler. curious why we need a switch statement to check the action type...